users/token/token.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Package token provides password recovery token generation with jwt
+// Token is sent by email to user as part of recovery URL
+// Token is signed by secret signature
+package token
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/users"
+)
+
+var (
+
+	// ErrMalformedToken malformed token
+	ErrMalformedToken = errors.New("Malformed token")
+	// ErrExpiredToken  password reset token has expired
+	ErrExpiredToken = errors.New("Token is expired")
+	// ErrWrongSignature wrong signature
+	ErrWrongSignature = errors.New("Wrong token signature")
+)
+
+const (
+	defTokenSecret   = "mainflux-secret"
+	defTokenDuration = "5"
+	defTokenLogLevel = "debug"
+
+	envTokenSecret   = "MF_TOKEN_SECRET"
+	envTokenDuration = "MF_TOKEN_DURATION"
+	envTokenLogLevel = "MF_TOKEN_DEBUG_LEVEL"
+)
+
+var once sync.Once
+
+type tokenizer struct {
+	hmacSampleSecret []byte // secret for signing token
+	tokenDuration    int    // token in duration in min
+	logger           logger.Logger
+}
+
+var t *tokenizer
+
+// Instance - Thread safe creation singleton instance of tokenizer.
+// Used for creating password reset token.
+func Instance() users.Tokenizer {
```

Comment:
Think about renaming it to `New()`. Also, do you need to force singleton for these structures? I think that there is no need for this (you will instantiate them in the `main.go` one anyway).